### ISSUE TITLE
False negative on the `Protocol` generic

### ISSUE THREAD

**Issue by belegnar** [microsoft/pyright#8158]
**Describe the bug**
`pyright` didn't report error on the `Protocol` generic

**Code or Screenshots**
Executing the code sample [pyright playground](https://pyright-play.net/?code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAK4MYAxmADYBQVpFAhgM6NQBCTApgFxVR9QImjGnSGEAFETAlyFANrtGHALoBKHvwFCqQA)

```python
from typing import Protocol

class Base:
    pass

class P(Protocol[Base]):
    pass

```
results in
```
TypeError: Parameters to Protocol[...] must all be type variables or parameter specification variables.
```

**VS Code extension or command-line**
Playground

**Comment by erictraut** [comment]
This isn't really a typing violation, but pyright does have checks for similar issues, so I think it's reasonable to add such a check. Changing from a bug report to an enhancement request.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.368.
